### ISSUE TITLE
Feature Request: allow config files to end with ".json5"

### ISSUE THREAD

**Issue by FelixGraf** [tusharmath/node-config-ts#84]
For roughly a month, it's possible to use JSON5 syntax in the config files, thanks to https://github.com/tusharmath/node-config-ts/pull/73. For code editors to understand that this is possible, allowing the config files to end with `.json5` as the file extension might make sense. 

When now using comments in the JSON file, both VS Code and WebStorm complain that this is a forbidden syntax. Having the file renamed to `default.json5` resolves the warnings.

If this is a desired feature, I will try to send a PR for it :)

<img width="334" alt="Bildschirm­foto 2023-02-03 um 15 57 32" src="https://user-images.githubusercontent.com/5724535/216635830-76eb220d-0076-455e-844f-8a114806ea17.png">

<img width="801" alt="Bildschirm­foto 2023-02-03 um 15 57 03" src="https://user-images.githubusercontent.com/5724535/216635837-4325a410-f6e8-46e8-bce8-b1085c7ede6e.png">

**Comment by tusharmath** [comment]
Would be awesome if you could send a PR!

**Comment by FelixGraf** [comment]
@tusharmath, on it at the moment :) 

But I want to clarify what should happen when there is both a `.json` and a `.json5` config file.

I see the following options:

1. merge both file contents (which one should take precedence?)
2. only parse one file and ignore the other one (which one should take precedence?)
3. throw an error explaining that there should be only one config file of the same type

Currently, I'm leaning toward implementing the third option, as it does not make much sense to have two separate files of the same type, but I'd love to hear your opinion!

**Comment by tusharmath** [comment]
I don't think we should throw errors. This is a fundamental problem with the current design of the library. You will face the same issue if you have a `.yml` file also. 
Have a look at #86 with the proposed new design. I think it can make life much easier and a design which is less opinionated and more robust.

**Comment by arthu** [comment]
It would be great if the library can support all format including .json, .json5 and yaml.
